cabal-install/src/Distribution/Client/Init/Types.hs

```diff
@@ -282,13 +289,23 @@ mkLiterate _ hs = hs
 -- -------------------------------------------------------------------- --
 -- Interactive prompt monad
 
+type PromptIO = ReaderT (Data.IORef.IORef SessionState) IO
```

Comment:
I'm not loving `type` aliases in general and for monads in particular. I'd use a `newtype`. But perhaps it's not a big deal here, and maybe it will just add noise. So, dunno.